### PR TITLE
Implemented interpolation search

### PR DESCRIPTION
**Pull request by ParthoKR**
Added interpolation search in go

### CODE REVIEW

**Comment by task4233** [comment]
@ParthoKR 

Would you remove `func main`?

See: #37

**Comment by ParthoKR** [comment]
> @ParthoKR
> 
> Would you remove `func main`?
> 
> See: #37

@task4233 main function was removed and renamed to InterpolationSearch

**Review by task4233 — COMMENTED** [review]
thx, @ParthoKR.
I've reviewed your code. Please check them out.
If you could find the time, I would like you to write test code.

File: searches/interpolationsearch.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package main
```
Comment (task4233) [inline searches/interpolationsearch.go:1]:
```suggestion
package interpolationsearch
```

**Comment by tjgurwara99** [comment]
Please take a look at our new [contribution guidelines](https://github.com/TheAlgorithms/Go/blob/d2f12849df76203d80762ceca5105531d3424384/CONTRIBUTING.md) and make changes accordingly. Currently this PR is not up to standard. If no changes are made in the next two weeks, I will consider this PR to be stale and will close this PR then.

**Pull request closed without merging**